sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1180,6 +1180,19 @@ object SQLConf {
       .booleanConf
       .createWithDefault(true)
 
+  object TypeCoercionMode extends Enumeration {
+    val default, hive = Value
```

Comment:
Hi, @wangyum .
What about using `val native` instead of `val default` here?
Then, I think you can remove unnecessary renaming like `~/native/binaryComparison.sql.out → ~/default/binaryComparison.sql.out`?